src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -270,7 +312,13 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
       // If we found reaching buffer size threshold or count based threshold, we will immediately
       // flush (Insert them)
       if (copiedStreamingBuffer != null) {
-        insertBufferedRecords(copiedStreamingBuffer);
+        if (this.enableSchematization && this.enableSchemaEvolution) {
+          insertBufferedRecordsWithRetry(
+              copiedStreamingBuffer,
+              SnowflakeSinkConnectorConfig.RETRY_INSERTION_ATTEMPT_MAX_COUNT_DEFAULT);
+        } else {
+          insertBufferedRecords(copiedStreamingBuffer);
+        }
```

Comment:
why we can't compute the retry count first and only call `insertBufferedRecordsWithRetry`?